include/seqan3/core/algorithm/configuration.hpp

```diff
@@ -0,0 +1,778 @@
+          typename fn_t>
+constexpr auto operator|(configuration_t && cfg,
+                         fn_t && fn)
+//!\cond
+    requires is_algorithm_configuration_v<remove_cvref_t<configuration_t>> &&
+             (is_type_specialisation_of_v<remove_cvref_t<fn_t>, configuration_fn_proxy> ||
+              std::is_base_of_v<configuration_fn_base<remove_cvref_t<fn_t>>, remove_cvref_t<fn_t>>)
+//!\endcond
+{
+    return fn(std::forward<configuration_t>(cfg));
+}
+
+/*!\brief Combines a configuration adaptor with another configuration adaptor.
+ * \tparam lhs_fn_t The configuration adaptor to be combined.
+ *                  Must be an instance of an instance of seqan3::detail::configuration_fn_base.
+ * \tparam rhs_fn_t Either an instance of seqan3::detail::configuration_fn_proxy or
+ *                  seqan3::detail::configuration_fn_base.
+ *
+ * \param[in] lhs_fn A configuration adaptor.
+ * \param[in] rhs_fn The functor, invoking the associated configuration adaptor.
+ * \returns The result of invoking the configuration adaptor with the passed `lhs_fn` object.
+ */
+template <typename lhs_fn_t,
+          typename rhs_fn_t>
+constexpr auto operator|(lhs_fn_t && lhs_fn,
+                         rhs_fn_t && rhs_fn)
+//!\cond
+    requires std::is_base_of_v<configuration_fn_base<remove_cvref_t<lhs_fn_t>>, remove_cvref_t<lhs_fn_t>> &&
+             (is_type_specialisation_of_v<remove_cvref_t<rhs_fn_t>, configuration_fn_proxy> ||
+             (std::is_base_of_v<configuration_fn_base<remove_cvref_t<rhs_fn_t>>, remove_cvref_t<rhs_fn_t>> &&
+              !std::is_same_v<remove_cvref_t<lhs_fn_t>, remove_cvref_t<rhs_fn_t>>))
+//!\endcond
+{
+    return rhs_fn(std::invoke(std::forward<lhs_fn_t>(lhs_fn), configuration<>{}));
+}
+
+/*!\brief Combines a seqan3::detail::configuration_fn_proxy adaptor with another configuration adaptor.
+ * \tparam proxy_fn_t The configuration adaptor proxy to be combined.
+ *                    Must be an instance of an instance of seqan3::detail::configuration_fn_proxy.
+ * \tparam rhs_fn_t   Either an instance of seqan3::detail::configuration_fn_proxy or
+ *                    seqan3::detail::configuration_fn_base.
+ *
+ * \param[in] proxy_fn A configuration adaptor proxy.
+ * \param[in] rhs_fn   The functor, invoking the associated configuration adaptor.
+ * \returns The result of invoking the configuration adaptor with the passed `proxy_fn` object.
+ */
+template <typename proxy_fn_t,
+          typename rhs_fn_t>
+constexpr auto operator|(proxy_fn_t && proxy_fn,
+                         rhs_fn_t && rhs_fn)
+//!\cond
+    requires is_type_specialisation_of_v<remove_cvref_t<proxy_fn_t>, configuration_fn_proxy> &&
+             (is_type_specialisation_of_v<remove_cvref_t<rhs_fn_t>, configuration_fn_proxy> ||
+             (std::is_base_of_v<configuration_fn_base<remove_cvref_t<rhs_fn_t>>, remove_cvref_t<rhs_fn_t>> &&
+              !std::is_same_v<remove_cvref_t<proxy_fn_t>, remove_cvref_t<rhs_fn_t>>))
```

Comment:
these requirements are quite hard to parse for humans, can they maybe be simplified by a concept somewhere?

Reply:
Maybe

Reply:
ok i removed it and joined both operator| overloads.